src/NuGet.Core/NuGet.ProjectModel/ProjectLockFile/PackagesLockFileUtilities.cs

```diff
@@ -72,6 +92,15 @@ public static bool IsLockFileStillValid(DependencyGraphSpec dgSpec, PackagesLock
                 }
             }
 
+            // Validate the runtimes for the current project did not change.
+            var projectRuntimesKeys = project.RuntimeGraph.Runtimes.Select(r => r.Key).Where(k => k != null);
+            var lockFileRuntimes = nuGetLockFile.Targets.Select(t => t.RuntimeIdentifier).Where(r => r != null);
```

Comment:
Do we need a distinct here? 

How does this behave when you have a multi targeting project.?

Reply:
Good catch! It should be distinct.